File: src/lib/pricing.ts
export const ITBIS_RATE = 0.18;

/**
 * El total siempre es la suma de líneas (mismo precio con o sin comprobante).
 * Con comprobante se calcula el ITBIS 18% incluido solo para mostrarlo.
 */
export function calcQuoteTotals(
  lineTotals: number[],
  includeItbis: boolean
): { subtotal: number; itbisAmount: number; total: number } {
  const total = round2(lineTotals.reduce((a, b) => a + b, 0));
  if (!includeItbis) {
    return { subtotal: total, itbisAmount: 0, total };
  }
  const itbisAmount = round2((total * ITBIS_RATE) / (1 + ITBIS_RATE));
  const subtotal = round2(total - itbisAmount);
  return { subtotal, itbisAmount, total };
}

export function round2(n: number): number {
  return Math.round((n + Number.EPSILON) * 100) / 100;
}

export function formatRD(amount: number): string {
  return new Intl.NumberFormat("es-DO", {
    style: "currency",
    currency: "DOP",
    minimumFractionDigits: 2,
  }).format(amount);
}

File: backend/internal/pricing/pricing.go
package pricing

const ITBISRate = 0.18

type Totals struct {
	Subtotal    float64 `json:"subtotal"`
	ITBISAmount float64 `json:"itbisAmount"`
	Total       float64 `json:"total"`
}

// CalcQuoteTotals keeps the same total with or without comprobante.
// When includeITBIS is true, ITBIS 18% is extracted for display only.
func CalcQuoteTotals(lineTotals []float64, includeITBIS bool) Totals {
	var sum float64
	for _, v := range lineTotals {
		sum += v
	}
	total := Round2(sum)
	if !includeITBIS {
		return Totals{Subtotal: total, ITBISAmount: 0, Total: total}
	}
	itbis := Round2(total * ITBISRate / (1 + ITBISRate))
	sub := Round2(total - itbis)
	return Totals{Subtotal: sub, ITBISAmount: itbis, Total: total}
}

func Round2(n float64) float64 {
	return float64(int(n*100+0.5)) / 100
}
